src/libopensc/card-openpgp.c

```diff
@@ -3551,6 +3561,15 @@ pgp_erase_card(sc_card_t *card)
 					"Card does not offer life cycle management");
 	}
 
+	if (r == SC_SUCCESS && priv->mf) {
+		new_mf = pgp_new_blob(card, NULL, priv->mf->id, priv->mf->file);
+		if (new_mf) {
+			pgp_free_blobs(priv->mf);
+			priv->mf = new_mf;
+			populate_blobs_to_mf(card, priv);
+		}
```

Comment:
I would propose to report some error when the allocation fails instead of silently skipping the rest of the code.

Something like this could do:
```suggestion
		new_mf = pgp_new_blob(card, NULL, priv->mf->id, priv->mf->file);
		if (new_mf == NULL) {
			LOG_TEST_RET(card->ctx, SC_ERROR_INTERNAL, "Failed to allocate the new MF blob");
		}
		pgp_free_blobs(priv->mf);
		priv->mf = new_mf;
		populate_blobs_to_mf(card, priv);
```